src/Aspire.Hosting/IDistributedApplicationBuilder.cs

```diff
@@ -136,6 +136,27 @@ public interface IDistributedApplicationBuilder
     [Experimental("ASPIREPIPELINES001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
     public IDistributedApplicationPipeline Pipeline { get; }
 
+    /// <summary>
+    /// Gets the service for managing Aspire directories.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// The <see cref="FileSystemService"/> provides a centralized way to manage directories
+    /// used by Aspire, including temporary files, cache, and other storage needs.
+    /// </para>
+    /// <para>
+    /// Resources and infrastructure code should use this service instead of static methods like
+    /// <see cref="Path.GetTempPath"/> or <see cref="Directory.CreateTempSubdirectory(string?)"/> to ensure
+    /// consistent directory management across the application.
+    /// </para>
+    /// <para>
+    /// Directory locations can be overridden using environment variables or configuration settings.
+    /// For example, the temp directory location can be overridden using the ASPIRE_TEMP_FOLDER
+    /// environment variable or the Aspire:TempDirectory configuration setting.
+    /// </para>
+    /// </remarks>
+    public IFileSystemService FileSystemService => throw new NotImplementedException();
```

Comment:
Should we expose this here or on context types in callbacks. This is a very loud location to put this API.

Reply:
Why not on IDAB?